enterprise/backend/test/metabase_enterprise/api/gsheets_test.clj

```diff
@@ -0,0 +1,211 @@
+(ns metabase-enterprise.api.gsheets-test
+  (:require
+   [clojure.java.io :as io]
+   [clojure.test :refer [deftest is testing]]
+   [java-time.api :as t]
+   [metabase-enterprise.gsheets.api :as gsheets.api]
+   [metabase-enterprise.harbormaster.client :as hm.client]
+   [metabase.test :as mt]
+   [toucan2.core :as t2])
+  (:import [java.time
+            LocalDate
+            LocalTime
+            ZoneId
+            ZonedDateTime]))
+
+(set! *warn-on-reflection* true)
+
+(deftest gsheets-calls-fail-when-missing-etl-connections
+  (mt/with-temporary-setting-values [api-key "some"]
+    (mt/with-premium-features #{:attached-dwh}
+      (is (= (str "ETL Connections is a paid feature not currently available to your instance. "
+                  "Please upgrade to use it. Learn more at metabase.com/upgrade/")
+             (:message (mt/user-http-request :crowberto :get 402 "ee/gsheets/service-account")))))))
+
+(deftest gsheets-calls-fail-when-not-activated
+  (mt/with-premium-features #{:etl-connections :attached-dwh}
+    (mt/with-temporary-setting-values [api-key nil]
+      (is (partial= {:message "Missing api-key."}
+                    (mt/user-http-request :crowberto :get 500 "ee/gsheets/service-account"))))))
+
+(deftest gsheets-calls-fail-when-missing-attached-dwh
+  (mt/with-temporary-setting-values [api-key "some"]
+    (mt/with-premium-features #{:etl-connections}
+      (is (= (str "Attached DWH is a paid feature not currently available to your instance. "
+                  "Please upgrade to use it. Learn more at metabase.com/upgrade/")
+             (:message (mt/user-http-request :crowberto :get 402 "ee/gsheets/service-account")))))))
+
+(deftest gsheets-calls-fail-when-non-superuser
+  (mt/with-premium-features #{:etl-connections :attached-dwh}
+    (mt/with-temporary-setting-values [api-key nil]
+      (is (= "You don't have permissions to do that."
+             (mt/user-http-request :rasta :get 403 "ee/gsheets/service-account"))))))
+
+(deftest gsheets-calls-fail-when-there-is-no-mb-api-key
+  (mt/with-premium-features #{:etl-connections :attached-dwh :hosting}
+    (mt/with-temporary-setting-values [api-key nil]
+      (is (= "Missing api-key."
+             (:message (mt/user-http-request :crowberto :get 500 "ee/gsheets/service-account")))))))
+
+(def happy-responses (read-string (slurp (io/resource "gsheets/mock_hm_responses.edn"))))
```

Comment:
gsheets/mock_hm_responses.edn is an edn map from {:method _ :url "" :body _} to the response observed from HM.

I've added a few functions like +syncing that overwrite pieces of the map with other observed responses.